VAMobile/src/screens/HealthScreen/LabsAndTests/LabsAndTestsList/LabsAndTestsListScreen.tsx

```diff
@@ -0,0 +1,310 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ScrollView } from 'react-native'
+
+import { StackScreenProps } from '@react-navigation/stack'
+
+import { DateTime } from 'luxon'
+import { map } from 'underscore'
+
+import { useLabsAndTests } from 'api/labsAndTests/getLabsAndTests'
+import { LabsAndTests } from 'api/types'
+import {
+  Box,
+  DefaultList,
+  DefaultListItemObj,
+  ErrorComponent,
+  FeatureLandingTemplate,
+  LoadingComponent,
+  Pagination,
+  PaginationProps,
+  TextLine,
+  TextView,
+  VAModalPicker,
+} from 'components'
+import { VAScrollViewProps } from 'components/VAScrollView'
+import { Events } from 'constants/analytics'
+import { DEFAULT_PAGE_SIZE } from 'constants/common'
+import { NAMESPACE } from 'constants/namespaces'
+import { ScreenIDTypesConstants } from 'store/api/types/Screens'
+import { a11yLabelVA } from 'utils/a11yLabel'
+import { logAnalyticsEvent } from 'utils/analytics'
+import { getA11yLabelText } from 'utils/common'
+import { formatDateMMMMDDYYYY } from 'utils/formattingUtils'
+import { getFormattedDate } from 'utils/formattingUtils'
+import { useError, useRouteNavigation, useTheme } from 'utils/hooks'
+import { screenContentAllowed } from 'utils/waygateConfig'
+
+import { HealthStackParamList } from '../../HealthStackScreens'
+import NoLabsAndTestsRecords from '../NoLabsAndTestsRecords/NoLabsAndTestsRecords'
+import { TimeFrameDropDownItem } from './TimeFrameType'
+import { TimeFrameTypeConstants } from './constants'
+
+type LabsAndTestsListScreenProps = StackScreenProps<HealthStackParamList, 'LabsAndTestsList'>
+
+/**
+ * Screen containing a list of Labs and Tests on record and a link to their details view
+ */
+function LabsAndTestsListScreen({ navigation }: LabsAndTestsListScreenProps) {
+  const theme = useTheme()
+  const { t } = useTranslation(NAMESPACE.COMMON)
+  const navigateTo = useRouteNavigation()
+  const [LabsAndTestsToShow, setLabsAndTestsToShow] = useState<Array<LabsAndTests>>([])
+
+  const [page, setPage] = useState(1)
+  const [datePickerOption, setDatePickerOption] = useState<TimeFrameDropDownItem>({} as TimeFrameDropDownItem)
+  const timeRangeOptions: Array<TimeFrameDropDownItem> = useMemo(() => {
+    const getMMMyyyy = (date: DateTime): string => {
+      return getFormattedDate(date.toISO(), 'MMM yyyy')
+    }
+
+    const getDateRange = (startDate: DateTime, endDate: DateTime): string => {
+      return `${getMMMyyyy(startDate)} - ${getMMMyyyy(endDate)}`
+    }
+    const todaysDate = DateTime.local()
+
+    const fiveMonthsEarlier = todaysDate.minus({ months: 5 }).startOf('month').startOf('day')
```

Comment:
This looks identical to whats in PastAppointment.tsx, can we abstract this out? If it's not exactly the same, can we move the `todaysDate.minus({ months: n }).startOf('month').startOf('day')` logic to a util?